src/Schema/Directives/RelationDirectiveHelpers.php

```diff
@@ -40,6 +40,7 @@ protected function makeBuilderDecorator($root, array $args, GraphQLContext $cont
             }
             assert($builder instanceof QueryBuilder || $builder instanceof EloquentBuilder);
 
+            /** @var QueryBuilder|EloquentBuilder<\Illuminate\Database\Eloquent\Model> $builder */
```

Comment:
should not be necessary - perhaps the type of `enhanceBuilder()` is fundamentally flawed?